hack/lib/build/environment.sh

```diff
@@ -36,15 +36,35 @@ function os::build::environment::create() {
     additional_context+=" -e OS_VERSION_FILE="
   fi
 
+  local cut=1
+  local env
+  env=( )
+  for arg in "${@:1}"; do
+    if [[ "${arg}" != *"="* ]]; then
```

Comment:
I was saying that this would mis-parse

```
VAR=val FOO=bar hack/my=filename.sh
```

But i don't think we need to handle that.